lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,37 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+
+# Remove or rename these "constants" in favor of a dynamic path
+use constant MAX_DURATION_SECONDS => 1800;
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app) {
+    path($ENV{OPENQA_GITLAB_MAINTENANCE_FILE} // ($app->config->{global}->{cache_dir} // '/var/lib/openqa/cache')
```

Comment:
The package has a generic name which I like but here you say "gitlab". Please keep it generic. And I don't think you need to set the complete path when the config isn't set. I assume we have other variables which define by default /var/lib/openqa?

Reply:
Yes i agree, though i'm not sure on how to word it, any better idea then `Source Control Management System` ? :sweat_smile:

Reply:
Well, we are within "Git::MaintenanceOutage". How about "git server outage"?

Reply:
By the way thinking about it I don't think we should call the package "MaintenanceOutage". We don't know why a remote server is not reachable. We can't assume it's due to a maintenance

Reply:
Changed to `OPENQA_GIT_SERVER_MAINTENANCE_FILE` and `OPENQA_GIT_SERVER_MAINTENANCE_DURATION`, is that better ?